app/controllers/pages/routes_controller.rb

```diff
@@ -3,4 +3,40 @@ def show
     back_link_url = form_pages_path(current_form.id)
     render locals: { current_form:, page:, pages: current_form.pages, back_link_url: }
   end
+
+  def delete
+    @delete_confirmation_input = Pages::Routes::DeleteConfirmationInput.new
+    render locals: { current_form:, page:, delete_confirmation_input: @delete_confirmation_input }
+  end
+
+  def destroy
+    @delete_confirmation_input = Pages::Routes::DeleteConfirmationInput.new(delete_confirmation_input_params)
+
+    if @delete_confirmation_input.invalid?
+      return render :delete, locals: { current_form:, page:, delete_confirmation_input: @delete_confirmation_input }
+    end
+
+    if @delete_confirmation_input.confirmed?
+      delete_routes
+      return redirect_to form_pages_path, success: t("banner.success.page_routes_deleted", question_position: current_form.page_number(page))
+    end
+
+    redirect_to show_routes_path(form_id: current_form.id, page_id: page.id)
+  end
+
+private
+
+  def delete_routes
+    all_form_routing_conditions = current_form.pages.flat_map(&:routing_conditions).compact_blank
+    page_routes = all_form_routing_conditions.select { |rc| rc.check_page_id == page.id }
+    page_routes.each do |rc|
+      rc.prefix_options[:form_id] = current_form.id
+      rc.prefix_options[:page_id] = rc.routing_page_id
+      ConditionRepository.destroy(rc)
+    end
```

Comment:
This doesn't feel like the best place for this. I'm not sure where is though?

We need to get the form, then every page and then delete the conditions.

It could go in one of the repositories but I'm not 100% sure which one is the best choice. 

It could be a separate service or a new RouteRepository itself?

Reply:
The idea of a route repository is interesting!

I wonder about putting it in the form input object though, that might make sense? I guess we would need a new one, but it could inherit from `DeleteConfirmationInput` to avoid too much code duplication?